### DISCUSSION TITLE
CoreDNS only has either AAAA or A records for services with Dual Stack enabled but not both.

### DISCUSSION THREAD

**Discussion by micksey9** [coredns/coredns#4893] [Q&A] [answered]
Hey all, I'm seeing some issues with core-dns when I'm creating services with both ipv4 and ipv6 addresses. Following is a sample service that I created and it gets both ipv6 and v4 addresses:

-- ---
[root@node1 k8_config]# kubectl describe svc my-nginx

Name:                     my-nginx
Namespace:                default
Labels:                   run=my-nginx
Annotations:              <none>
Selector:                 run=my-nginx
Type:                     LoadBalancer
**IP Family Policy:         RequireDualStack
IP Families:              IPv6,IPv4**
IP:                       fd85:ee78:d8a6:8607::19eb
IPs:                      fd85:ee78:d8a6:8607::19eb,10.233.44.168
LoadBalancer Ingress:     1000::e
Port:                     <unset>  80/TCP
TargetPort:               80/TCP
NodePort:                 <unset>  31854/TCP
Endpoints:                [fd85:ee78:d8a6:8607::1:200a]:80,[fd85:ee78:d8a6:8607::1:200b]:80
Session Affinity:         None
External Traffic Policy:  Cluster

-- ---

But I only see AAAA record for the above service while query for A record returns nothing:

[root@node1 test]# dig @10.233.0.3 my-nginx.default.svc.k8cluster.local AAAA +short
**fd85:ee78:d8a6:8607::19eb**

[root@node1 test]# dig @10.233.0.3 my-nginx.default.svc.k8cluster.local A +short
[root@node1 test]#

Now if I change the ipFamily parameters for my-nginx service to the following:

**IP Family Policy:         RequireDualStack
IP Families:              IPv4,IPv6**

Then Core-DNS only has A record for the service and no AAAA.

Is there some additional config that is required for Core-DNS to generate both A and AAAA records?

**Comment by chrisohaver** [comment, marked as answer]
What version of CoreDNS are you using?
CoreDNS doesn't support dual stack services until 1.8.1.

**Reply by micksey9** [reply]
Ahh, I did my cluster install through kubespray and that has Kubernetes version 1.21 so CoreDNS is at 1.8.0. I will upgrade to the latest and test with that, thanks.

**Reply by chrisohaver** [reply]
OK. Note that CoreDNS >= 1.8.1 must be granted list and watch access to endpointslices.

https://github.com/coredns/deployment/blob/292169955ccc2d670860ac4cfc90e0fb644c55b3/kubernetes/coredns.yaml.sed#L24-L30
